test/framework/src/main/java/org/elasticsearch/transport/MockTcpTransport.java

```diff
@@ -294,26 +324,21 @@ public void onFailure(Exception e) {
                 @Override
                 protected void doRun() throws Exception {
                     StreamInput input = new InputStreamStreamInput(new BufferedInputStream(activeChannel.getInputStream()));
-                    while (isOpen.get()) {
+                    while (isOpen.get() && !Thread.currentThread().isInterrupted()) {
```

Comment:
Our test infrastructure for example. Admittedly, this is a *really* small window and very unlikely, but it is possible nevertheless.